arangod/Graph/Cursors/RefactoredSingleServerEdgeCursor.cpp

```diff
@@ -269,12 +306,20 @@ void RefactoredSingleServerEdgeCursor<Step>::readAll(
     auto& cursor = lookupInfo.cursor();
     LogicalCollection* collection = cursor.collection();
     auto cid = collection->id();
-    bool hasExtra = !_requiresFullDocument && cursor.hasExtra();
     auto* expression = lookupInfo.getExpression();
+    uint16_t coveringPosition = lookupInfo.coveringIndexPosition();
 
-    if (hasExtra) {
-      cursor.allExtra([&](LocalDocumentId const& token, VPackSlice edge) {
+    if (!_requiresFullDocument &&
+        coveringPosition != aql::Projections::kNoCoveringIndexPosition) {
```

Comment:
Moving that statement into a seperate function would help to maintain readability here. (From a technical point it is not required, but IMHO it would help improve the readability. Just by reading that comparison (without further knowledge) it is hard to understand (e.g. new developers).